src/libmongoc/tests/unified/runner.c

```diff
@@ -276,24 +282,22 @@ static test_t *
 test_new (test_file_t *test_file, bson_t *bson)
 {
    test_t *test;
+   bson_parser_t *parser;
 
    test = bson_malloc0 (sizeof (test_t));
    test->test_file = test_file;
-   test->description = bson_lookup_utf8 (bson, "description");
-   if (bson_has_field (bson, "runOnRequirements")) {
-      test->run_on_requirements = bson_lookup_bson (bson, "runOnRequirements");
-   }
-   if (bson_has_field (bson, "skipReason")) {
-      test->skip_reason = bson_lookup_utf8 (bson, "skipReason");
-   }
-   test->operations = bson_lookup_bson (bson, "operations");
-   if (bson_has_field (bson, "expectEvents")) {
-      test->expect_events = bson_lookup_bson (bson, "expectEvents");
-   }
-   if (bson_has_field (bson, "outcome")) {
-      test->outcome = bson_lookup_bson (bson, "outcome");
-   }
-
+   parser = bson_parser_new ();
+   bson_parser_utf8 (parser, "description", &test->description);
+   bson_parser_array_optional (
+      parser, "runOnRequirements", &test->run_on_requirements);
+   bson_parser_utf8_optional (parser, "skipReason", &test->skip_reason);
+   bson_parser_array_optional (parser, "operations", &test->operations);
```

Comment:
Isn't this required? From [here](https://github.com/mongodb/specifications/blob/master/source/unified-test-format/unified-test-format.rst#test):

> operations: Required array of one or more operation objects. List of operations to be executed for the test case.

Reply:
Ah good catch, done.